### DISCUSSION TITLE
Paper: Mutual Reasoning Makes Smaller LLMs Stronger Problem-Solvers

### DISCUSSION THREAD

**Discussion by blackeuler** [daveshap/Raspberry#43] [Research]
This may be of interest for a method of data generation for reasoning steps. 
https://arxiv.org/abs/2408.06195
https://github.com/zhentingqi/rStar

The prompts seem worthy of further analysis to how we might want to template real world tasks and reasoning traces.

**Comment by tinycrops** [comment]
this is an interesting related work [Unexpected Benefits of Self-Modeling in Neural Systems](https://arxiv.org/abs/2407.10188) as self-modeling is a core component of effective mutual modeling.
"Mutual modelling, the reciprocal ability to establish a mental model of the other, plays a fundamental role in human
interactions. This complex cognitive skill is however difficult to fully apprehend as it encompasses multiple neuronal,
psychological and social mechanisms that are generally not
easily turned into computational models suitable for robots.
This article presents several perspectives on mutual modelling from a range of disciplines, and reflects on how these
perspectives can be beneficial to the advancement of social
cognition in robotics." [Mutual Modelling in Robotics: Inspirations for the Next Steps](https://www.researchgate.net/publication/283675963_Mutual_Modelling_in_Robotics_Inspirations_for_the_Next_Steps)
![image](https://github.com/user-attachments/assets/84c4b296-053e-4942-88a9-a0e8a6b748cb)
